### PR TITLE
Fixes a minor deprecation warning

### PR DESCRIPTION
**Pull request by res-life**
This fixes https://github.com/NVIDIA/spark-rapids/issues/3374

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
**1. getAllStatistics deprecated**
```
[WARNING] [Warn] GpuMultiFileReader.scala:87: 
method getAllStatistics in class FileSystem is deprecated: 
see corresponding Javadoc for more information.
```
This is because of getAllStatistics is deprecated in hadoop-client-api-3.3.1.jar which is depended by spark 3.2.0.
Recommends to use 

see: 
```
https://hadoop.apache.org/docs/r2.8.2/api/org/apache/hadoop/fs/FileSystem.html
```
But getGlobalStorageStatistics does not provide the capacity to fetch thread local statistics like previous version:
```
FileSystem.getAllStatistics.asScala.map(_.getThreadStatistics.getBytesRead).sum
```
So, just leave it as it is. BTW, one issue is filed: https://github.com/NVIDIA/spark-rapids/issues/4219
**2. getDecimalMetadata deprecated**
```
[WARNING] [Warn] sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala:604: 
method getDecimalMetadata in class PrimitiveType is deprecated: 
see corresponding Javadoc for more information.
```
spark 3.0.1 uses parquet-column-1.10.1.jar and spark 3.2.0 uses parquet-column-1.12.1.jar.
parquet-column 1.12.1 deprecated the getDecimalMetadata method and recommends to use "getLogicalTypeAnnotation()".
parquet-column-1.10.1 does not have getLogicalTypeAnnotation.
So if want to remove the warning we must write shim codes separately for the 2 versions.
It's a little heavy, let's keep it.

**3.  isDirectory deprecated**
```
sql-plugin/src/main/scala/org/apache/spark/sql/execution/datasources/rapids/GpuPartitioningUtils.scala:121:
 method isDirectory in class FileSystem is deprecated: 
see corresponding Javadoc for more information.
```
Fixed.

**Comment by res-life** [comment]
build

**Review by GaryShen2008 — APPROVED** [review]

File: sql-plugin/src/main/scala/org/apache/spark/sql/execution/datasources/rapids/GpuPartitioningUtils.scala [hunk 1]
```diff
@@ -121,7 +118,7 @@ object GpuPartitioningUtils {
     basePathOption match {
       case Some(userDefinedBasePath) =>
         val fs = userDefinedBasePath.getFileSystem(hadoopConf)
-        if (!fs.isDirectory(userDefinedBasePath)) {
+        if (!fs.getFileStatus(userDefinedBasePath).isDirectory) {
```
Comment (firestarman) [inline sql-plugin/src/main/scala/org/apache/spark/sql/execution/datasources/rapids/GpuPartitioningUtils.scala:121]: Is it possible to return a `null` from `getFileStatus` ?

Comment (res-life) [reply sql-plugin/src/main/scala/org/apache/spark/sql/execution/datasources/rapids/GpuPartitioningUtils.scala:121]: Throws FileNotFoundException, its javadoc:
    FileNotFoundException – when the path does not exist;

**Pull request merged**